x/dex/types/message_multi_hop_swap.go

```diff
@@ -70,6 +70,20 @@ func (msg *MsgMultiHopSwap) ValidateBasic() error {
 		return ErrMissingMultihopRoute
 	}
 
+	for _, route := range msg.Routes {
```

Comment:
A bit of a nitpick, but we're already iterating through the routes below, would be a bit cleaner to combine all checks into a single loop.